tests/test_async_patterns.py

```
@@ -0,0 +1,546 @@
+import os
+import sys
+import time
+import unittest
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+
+# Add src to path for imports
+sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "src"))
+
+from mmrelay.matrix_utils import connect_matrix, matrix_relay, on_room_message
```

Ruff (0.12.2)

`mmrelay.matrix_utils.on_room_message` imported but unused

Remove unused import: `mmrelay.matrix_utils.on_room_message`

(F401)